lib/dataset/groupby.cpp

```diff
@@ -166,27 +169,45 @@ template <class T> T GroupBy<T>::concat(const Dim reductionDim) const {
 
 /// Reduce each group using `sum` and return combined data.
 template <class T> T GroupBy<T>::sum(const Dim reductionDim) const {
-  return reduce(sum_impl, reductionDim, FillValue::ZeroNotBool);
+  return reduce(sum_into, reductionDim, FillValue::ZeroNotBool);
 }
 
 /// Reduce each group using `all` and return combined data.
 template <class T> T GroupBy<T>::all(const Dim reductionDim) const {
-  return reduce(all_impl, reductionDim, FillValue::True);
+  return reduce(
+      [](Variable &out, const Variable &var) {
+        accumulate_in_place(out, var, core::element::logical_and_equals,
+                            "groupby.all");
+      },
+      reductionDim, FillValue::True);
```

Comment:
This lambda looks like `reduce_impl`?

Reply:
Yes. I wanted to remove that link between the `operations_common.h` and the dataset module. That is why I inlined the `all_impl` etc functions here. Since `reduce_impl` only forwards arguments, I saw no reason to expose it to `dataset` and use it here.

Reply:
Maybe worth defining a helper function here then?